# Lab | Inferential statistics - T-test & P-value

**Instructions:**

    We will have another simple example on two sample t test (pooled - when the variances are equal). But this time this is a one sided t-test.

In a packing plant, a machine packs cartons with jars. It is supposed that a new machine will pack faster on the average than the machine currently used. To test that hypothesis, the times it takes each machine to pack ten cartons are recorded. The results, in seconds, are shown in the tables in the file files_for_lab/machine.txt. Assume that there is sufficient evidence to conduct the t test, does the data provide sufficient evidence to show if one machine is better than the other?


In [1]:
import pandas as pd
import numpy as np
pd.set_option('display.max_columns', None)
import warnings
warnings.filterwarnings('ignore')
import matplotlib.pyplot as plt
import seaborn as sns
import math

In [2]:
import codecs

machine = codecs.open('files_for_lab/machine.txt', 'r', 'utf-16')
machine = pd.read_csv(machine, sep = '\t')
machine.columns = ['new_machine', 'old_machine']
machine

new_machine  old_machine
0         42.1         42.7
1         41.0         43.6
2         41.3         43.8
3         41.8         43.3
4         42.4         42.5
5         42.8         43.5
6         43.2         43.1
7         42.3         41.7
8         41.8         44.0
9         42.7         44.1

In [3]:
student = codecs.open('files_for_lab/student_gpa.txt', 'r', 'utf-8')
student = pd.read_csv(student, sep = '\t')
student.columns = ['sophomores', 'juniors']
student

sophomores  juniors
0         3.04     2.56
1         1.71     2.77
2         3.30     2.70
3         2.88     3.00
4         2.11     2.98
5         2.60     3.47
6         2.92     3.26
7         3.60     3.20
8         2.28     3.19
9         2.82     2.65
10        3.03     3.00
11        3.13     3.39
12        2.86     2.58
13        3.49      NaN
14        3.11      NaN
15        2.13      NaN
16        3.27      NaN

In [4]:
avg_new_machine = np.mean(machine['new_machine'])
avg_new_machine

42.14

In [5]:
avg_old_machine = np.mean(machine['old_machine'])
avg_old_machine

43.230000000000004

In [6]:
sample_mean1 = avg_new_machine
sample_std1 = np.std(machine['new_machine'])
n1 = 10
sample_mean2 = avg_old_machine
sample_std2 = np.std(machine['old_machine'])
n2 = 10

pooled_sample_std = math.sqrt(((n1-1)*sample_std1**2 + (n2-1)*sample_std2**2)/(n1+n2-2))
statistic = (sample_mean1-sample_mean2)/(pooled_sample_std*math.sqrt((1/n1)+(1/n2)))
print("T Statistic is: ", statistic)

T Statistic is:  -3.5809955894645813


In [7]:
from scipy.stats import t

print("P value is: ", t.cdf(statistic,n1+n2-2))
print("Critical Value of z is: ", t.ppf(0.05, n1+n2-2))
#alpha is 0.05 (one-sided!)

P value is:  0.0010677934016937686
Critical Value of z is:  -1.734063606617536


    - H0: no difference between new and old machine
    - Ha: new machine is faster or slower than old one
    - confidence level: 0.95
    - degress of freedom: 10 + 10 - 2 = 18
    - critical value = 1.734
    - we reject the null hypothesis (so new machine is faster or slower)